temporal-sdk/src/main/java/io/temporal/internal/worker/PollTaskExecutor.java

```diff
@@ -91,6 +100,8 @@ public void process(T task) {
             MDC.remove(LoggerTag.NAMESPACE);
             MDC.remove(LoggerTag.TASK_QUEUE);
           }
+          // this thread will be released right after, so updating count with +1 adjuster
+          publishSlotsMetric(1);
```

Comment:
I don't believe this is a thread-safe approach. If there are 10 slots and 5 being used and all 5 finish at the same time calling this, the value might end up at 10 - 5 + 1 racily, which is 6 when really 10 will be available. If you don't want to maintain a counter that you incr/decr, is there somewhere _after_ the task executes that you refresh the number?

Reply:
It's not a thread-safe approach at all. It's a good enough approach. See "Additional context" here:  https://github.com/temporalio/sdk-java/issues/998
> is there somewhere after the task executes that you refresh the number?
no, there is none right now.